### PR TITLE
feat(getValueDoppelganger): Ignore css property values with `var()`

### LINKED ISSUE #61: css vars being transformed incorrectly with `textShadow` converter
- `rtl-css-js` version: 1.14.1
- `node` version: 14
- `npm` (or `yarn`) version: 1.22.x

Relevant code or config

```javascript
convertProperty('boxShadow', 'var(--shadow16)');
```

What you did:

Fluent UI's css-in-js engine uses rtl-css-js for its rtl property and value computations. I noticed that css variable names that included numbers were matched by the `textShadow` property value converter and transformed the name of the css variable

What happened:

`var(--shadow16)` became `var(--shadow-16)` which broke all rtl box shadows using Fluent UI.

Reproduction repository:
https://github.com/ling1726/box-shadow-rtl-css-js-repro

Problem description:

See above in `What you did`

Suggested solution:

This library could consider ignoring rtl doppelganger computations for any value that includes css variables. This is already the case for undefined and boolean values.

I'm also open to push back, if this check should be done by consumers.

### PR DESCRIPTION
**Pull request by ling1726**
**What**:

Fixes #61 which illustrates a concrete example of the problem

Since the original issue is only with `boxShadow` or `textShadow` I'm
open to any suggestions that move the `containsCssVar` check only for
that property value converter.

**How**:

Created an extra `containsCssVar` regex match and used it with the existing checks
for boolean and undefined property values. Extracted the check to a separate utility
to avoid too much complexity in the function.

I was initially worried about some edge cases:

*  `translateX(var())`
* `padding: '2px var(--foo)';`

but after testing, it seems that the results are the same before and after the changes in this PR.

**Checklist**:

- [x] Documentation
- [x] Tests
- [x] Ready to be merged
      
- [x] Added myself to contributors table

### CODE REVIEW

File: src/__tests__/index.js [hunk 1]
```diff
@@ -431,6 +431,12 @@ const unchanged = [
   [{opacity: () => {}}],
   [{objectPosition: 'center bottom'}],
   [{objectPosition: '5px 10px'}], // There's no RTL-flipped equivalent for the 5px. :-(
+  [{boxShadow: 'var(--shadow16)'}],
```
Comment (ling1726) [inline src/__tests__/index.js:434]: with the exception of `boxShadow` the below tests were all passing before and after the changes in this PR

**Comment by kentcdodds** [comment]
Thanks for this! I hadn't thought about css variables. I wonder if it would ever make sense for someone to have two different variables assigned and hope that they would get swapped for them: `var(--padding-left-lg)` -> `var(--padding-right-lg)`

Do you think there'd be a way to support this?

**Comment by ling1726** [comment]
> Thanks for this! I hadn't thought about css variables. I wonder if it would ever make sense for someone to have two different variables assigned and hope that they would get swapped for them: `var(--padding-left-lg)` -> `var(--padding-right-lg)`
> 
> Do you think there'd be a way to support this?

In the scope of a library that does transforms, that seems to me like some shaky logic. Wouldn't it just be string replacing any variable we see with `right`-> `left` and vice versa ? 

Perhaps I'm misunderstanding you, an example might help in that case

**Comment by kentcdodds** [comment]
Nah, you're right. That would not be advisable to rely on. Let's go with this as a bug fix 👍

**Review by kentcdodds — APPROVED** [review]
Thanks!

**Comment by ling1726** [comment]
@kentcdodds thx for merging, just wondering when this would be released?

Also did I add myself to contributors correctly?

**Comment by kentcdodds** [comment]
@all-contributors please add @ling1726 for code and tests

**Comment by kentcdodds** [comment]
Sorry about that. I'm not sure why a release didn't go out and I forgot to fix the contributor thing. I use a bot so you don't have to add yourself manually :)

**Comment by ling1726** [comment]
Thx! Couldn't ask for more for a first time oss contribution

**Comment by kentcdodds** [comment]
We're having trouble with the release: https://github.com/kentcdodds/rtl-css-js/runs/4197260969?check_suite_focus=true#step:7:137

I'm not sure why this is happening 😬 I'll figure out how to get it fixed though.

**Comment by kentcdodds** [comment]
Fixed and released!
